src/components/MyObservations/MyObservations.js

```diff
@@ -0,0 +1,166 @@
+// @flow
+import Header from "components/MyObservations/Header";
+import ViewWithFooter from "components/SharedComponents/ViewWithFooter";
+import { View } from "components/styledComponents";
+import type { Node } from "react";
+import React, { useRef, useState } from "react";
+import {
+  Animated, Dimensions, Platform
+} from "react-native";
+import useCurrentUser from "sharedHooks/useCurrentUser";
+import useLocalObservations from "sharedHooks/useLocalObservations";
+import useUploadObservations from "sharedHooks/useUploadObservations";
+
+import InfiniteScrollLoadingWheel from "./InfiniteScrollLoadingWheel";
+import LoginSheet from "./LoginSheet";
+import MyObservationsEmpty from "./MyObservationsEmpty";
+import MyObservationsPressable from "./MyObservationsPressable";
+import ObsGridItem from "./ObsGridItem";
+import ObsListItem from "./ObsListItem";
+
+const { diffClamp } = Animated;
+
+type Props = {
+  isLoading?: bool,
+  layout: "list" | "grid",
+  observations: Array<Object>,
+  onEndReached: Function,
+  setLayout: Function
+}
+
+const {
+  width: screenWidth,
+  height: screenHeight
+} = Dimensions.get( "screen" );
+const GUTTER = 15;
+
+const MyObservations = ( {
+  isLoading,
+  layout,
+  observations,
+  onEndReached,
+  setLayout
+}: Props ): Node => {
+  const currentUser = useCurrentUser( );
+  const [heightAboveToolbar, setHeightAboveToolbar] = useState( 0 );
+
+  const [hideHeaderCard, setHideHeaderCard] = useState( false );
+  const [yValue, setYValue] = useState( 0 );
+  // basing collapsible sticky header code off the example in this article
+  // https://medium.com/swlh/making-a-collapsible-sticky-header-animations-with-react-native-6ad7763875c3
+  const scrollY = useRef( new Animated.Value( 0 ) );
+  const scrollYClamped = diffClamp( scrollY.current, 0, heightAboveToolbar );
+
+  const offsetForHeader = scrollYClamped.interpolate( {
+    inputRange: [0, heightAboveToolbar],
+    // $FlowIgnore
+    outputRange: [0, -heightAboveToolbar]
+  } );
+
+  const setNumColumns = ( ) => {
+    if ( layout === "list" || screenWidth <= 320 ) { return 1; }
+    if ( screenWidth <= 744 ) { return 2; }
+    if ( screenWidth <= 1024 ) { return 4; }
+    return 6;
+  };
+
+  const numColumns = setNumColumns( );
+  const combinedGutterWidth = ( numColumns + 1 ) * GUTTER;
+  const gridItemWidth = Math.round( ( screenWidth - combinedGutterWidth ) / numColumns );
+
+  const { allObsToUpload } = useLocalObservations( );
+  const uploadStatus = useUploadObservations( allObsToUpload );
```

Comment:
These two values should be pulled up into the container because they don't concern presentation.